### DISCUSSION TITLE
Significance of 0xb4c79daB8f259C7Aee6E5b2Aa729821864227e84 mentioned in foundry doc

### DISCUSSION THREAD

**Discussion by devrajsinghrawat** [Cyfrin/foundry-full-course-cu#548] [Q&A]
Hello Friends,

Did anyone understands the significance of address 0xb4c79daB8f259C7Aee6E5b2Aa729821864227e84 which is mentioned in https://book.getfoundry.sh/forge/writing-tests 

Tests are deployed to 0xb4c79daB8f259C7Aee6E5b2Aa729821864227e84. If you deploy a contract within your test, then 0xb4c...7e84 will be its deployer. If the contract deployed within a test gives special permissions to its deployer, such as Ownable.sol's onlyOwner modifier, then the test contract 0xb4c...7e84 will have those permissions.

**Comment by shikhar229169** [comment]
When you run test then the Test contract is deployed at address 0xb4c79daB8f259C7Aee6E5b2Aa729821864227e84. The address is just taken as example and it can be address.

And then within the test contract if you deploy any another contract (say XYZ) then the deployer of that contract XYZ will be the address of Test contract, because over here the contract XYZ was deployed by the Test contract. And considering their example it will be this address - 0xb4c79daB8f259C7Aee6E5b2Aa729821864227e84.

**Reply by devrajsinghrawat** [reply]
Ok If that is just an example address then it is just an address and not an environment setup address.. Btw for me the Test Contract address is consistent 0x7FA9385bE102ac3EAc297483Dd6233D62b3e1496 

Is this the same case for you guys?

**Reply by shikhar229169** [reply]
Yes.

**Comment by theirrationalone** [comment]
We can say the address/deployer is the foundry who has all permissions to test our contracts... 

Just as in hardhat where we pick deployer: 0 as default who deploys all the contracts in test env. through "getNamedAccounts() [0]" (If returns an array). 

Ownable modifier might helps to make address(this) the deployer if any contract deployed independently inside another contract... I'm not sure here <-
